FWCore/Common/interface/OutputProcessBlockHelper.h

```diff
@@ -0,0 +1,76 @@
+#ifndef FWCore_Common_OutputProcessBlockHelper_h
+#define FWCore_Common_OutputProcessBlockHelper_h
+
+/** \class edm::OutputProcessBlockHelper
+
+\author W. David Dagenhart, created 28 December, 2020
+
+*/
+
+#include "DataFormats/Provenance/interface/ProvenanceFwd.h"
+
+#include <set>
+#include <string>
+#include <vector>
+
+namespace edm {
+
+  class ProcessBlockHelperBase;
+
+  class OutputProcessBlockHelper {
+  public:
+    std::vector<std::string> const& processesWithProcessBlockProducts() const {
+      return processesWithProcessBlockProducts_;
+    }
+
+    void updateAfterProductSelection(std::set<std::string> const& processesWithKeptProcessBlockProducts,
+                                     ProcessBlockHelperBase const&);
+
+    void fillCacheIndices(StoredProcessBlockHelper&) const;
+
+    bool productsFromInputKept() const { return productsFromInputKept_; }
+
+    static constexpr unsigned int invalidCacheIndex() { return 0xffffffff; }
```

Comment:
Thanks for noticing this. After looking at this more closely, I realize that we only need the functions invalidCacheIndex and invalidProcessIndex defined in ProcessBlockHelperBase. The ones in StoredProcessBlockHelper and OutputProcessBlockHelper can be deleted. I am implementing this change now and it will be in the next set of commits I push.

StoredProcessBlockHelper does not need to know these special values at all. In an earlier version, I had some non-trivial functions in StoredProcessBlockHelper. But I moved those out of that class because it is a DataFormat. And it is not a problem for OutputProcessBlockHelper.cc to use the value from ProcessBlockHelperBase. It already includes the header. Much better to remove this duplication.